### DISCUSSION TITLE
NavigationStack problem

### DISCUSSION THREAD

**Discussion by easiwriter** [pointfreeco/swift-composable-architecture#3456] [General]
Thanks for your reply to an earlier post (CPU Overload).  I realised that I was missing public init functions for the reducers, so I added those plus some more elsewhere and now the whole project compiles except for one situation. The attached zip contains the details.

[Archive.zip](https://github.com/user-attachments/files/17451207/Archive.zip)

RootFeature contains the reducer and view for the root. The error report applies to the RootView starting at line 63. The files RootFeature 2 and RootView are from the unmodularised version of the project which compiles without error. The only difference is the addition of public in the failing version.

I'm sure the answer to this is obvious. I just can't see it.

**Comment by easiwriter** [comment]
I removed RootFeature and RootView from the package only to find they still don't compile, so I commented out some of the code as shown:

<img width="850" alt="Screenshot 2024-10-22 at 09 34 56" src="https://github.com/user-attachments/assets/40343b59-1ae7-41f1-a9f0-6f2031d8b1e2">

Removing ProjectListView causes the non-function type error, removing the FoldersListView cause the scope error. At the moment I can't see why this is happening. Something about these other modules is upsetting things.

**Comment by stephencelis** [comment]
@easiwriter The zip you attached doesn't appear to contain a project, just a few source files. Can you share an entire project that's ready to compile?

**Reply by easiwriter** [reply]
I've fixed it - the problem was obvious: neither ProjectListView nor FoldersListView are declared public. That said I don't understand where the compilation errors come from. They certainly don't point to the actual problem!
